flink-clients/src/main/java/org/apache/flink/client/cli/ExecutionConfigAccessor.java

```diff
@@ -61,7 +62,10 @@ public static ExecutionConfigAccessor fromProgramOptions(final ProgramOptions op
 
 		final Configuration configuration = new Configuration();
 
-		configuration.setInteger(CoreOptions.DEFAULT_PARALLELISM, options.getParallelism());
+		if (options.getParallelism() != ExecutionConfig.PARALLELISM_DEFAULT) {
```

Comment:
Does this commit go into a separated story?

Reply:
I will squash them when merging. It was just for the reviewing.